src/host-session-service.vala

```diff
@@ -345,6 +345,29 @@ namespace Frida {
 			throw new Error.INVALID_OPERATION ("Only meant to be implemented by services");
 		}
 
+		public async HashTable<string, Variant>? query_system_parameters (Cancellable? cancellable) throws Error, IOError {
+			#if DARWIN && !IOS
+				Variant? platform = new Variant ("s", "macos");
+			#elif IOS
+				Variant? platform = new Variant ("s", "ios");
+			#elif ANDROID && (X86 || X86_64)
+				Variant? platform = new Variant ("s", "android_emulated");
```

Comment:
This is wrong in some cases, like with Android x86 tablets, so we shouldn't try to disambiguate here.